src/graphics/shader.cpp

```diff
@@ -0,0 +1,119 @@
+//
+// Created by Lukasz Gut on 22/12/2018.
+//
+
+#include "shader.h"
+#include <glad/glad.h>
+#include "pch.h"
+#include "file_manager.h"
+
+namespace Bald::Graphics {
+
+    Shader::Shader(const char *vertexPath, const char *fragmentPath)
+    : m_ShaderID(0), m_VertexPath(vertexPath), m_FragmentPath(fragmentPath) {
+        m_ShaderID = CreateShader();
+    }
+
+    Shader::~Shader() {
+        glDeleteProgram(m_ShaderID);
+    }
+
+    void Shader::Bind() const {
+        glUseProgram(m_ShaderID);
+    }
+
+    void Shader::Unbind() const {
+        glUseProgram(0);
+    }
+
+    void Shader::SetUniform4f(const char* uniformName, float v0, float v1, float v2, float v3) const {
+        glUniform4f(GetUniformLocation(uniformName), v0, v1, v2, v3);
+    }
+
+    inline unsigned int Shader::GetShaderID() const {
+        return m_ShaderID;
+    }
+
+    unsigned int Shader::CreateShader() const {
+        using namespace Bald;
+        int success;
+
+        std::string vertexCodeString = Utils::FileManager::ReadFile(m_VertexPath, Utils::FileManager::SMALL_FILE);
+        std::string fragmentCodeString = Utils::FileManager::ReadFile(m_FragmentPath, Utils::FileManager::SMALL_FILE).c_str();
+
+        const char* vertexCode = vertexCodeString.c_str();
+        const char* fragmentCode = fragmentCodeString.c_str();
+
+        unsigned int vertexShader = glCreateShader(GL_VERTEX_SHADER);
+        glShaderSource(vertexShader, 1, &vertexCode, nullptr);
+        glCompileShader(vertexShader);
+
+        glGetShaderiv(vertexShader, GL_COMPILE_STATUS, &success);
+        if(!success) {
```

Comment:
I think we should talk about convention of conditions in if statement because in my opinion success == GL_FALSE is more readable and giving us instant information what is variable success

Reply:
Agree, fixed.